e2e/cephfs.go

```diff
@@ -261,6 +261,52 @@ var _ = Describe("cephfs", func() {
 				}
 			})
 
+			By("create PVC in storageClass with volumeNamePrefix", func() {
+				volumeNamePrefix := "foo-bar-"
+				err := createCephfsStorageClass(f.ClientSet, f, false, map[string]string{"volumeNamePrefix": volumeNamePrefix})
+				if err != nil {
+					e2elog.Failf("failed to create storageclass with error %v", err)
+				}
+				// set up PVC
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC with error %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+				err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC with error %v", err)
+				}
+
+				validateSubvolumeCount(f, 1, fileSystemName, subvolumegroup)
+				// list subvolumes and check if one of them has the same prefix
+				foundIt := false
+				subvolumes, err := listCephFSSubVolumes(f, fileSystemName, subvolumegroup)
+				if err != nil {
+					e2elog.Failf("failed to list subvolumes with error %v", err)
+				}
+				for _, subVol := range subvolumes {
+					fmt.Printf("Checking prefix on %s\n", subVol)
+					if strings.HasPrefix(subVol.Name, volumeNamePrefix) {
+						foundIt = true
+						break
+					}
+				}
+				// clean up after ourselves
+				err = deletePVCAndValidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to  delete PVC with error %v", err)
+				}
+				validateSubvolumeCount(f, 0, fileSystemName, subvolumegroup)
+				err = deleteResource(cephfsExamplePath + "storageclass.yaml")
+				if err != nil {
+					e2elog.Failf("failed to delete storageclass with error %v", err)
+				}
+				if !foundIt {
```

Comment:
Why not move this check just after the for loop?

Reply:
just to clean up the pvc and storageclass created as part of this test same pattern is followed in RBD as well https://github.com/ceph/ceph-csi/blob/master/e2e/rbd.go#L771-L827